selfdrive/ui/update_translations.py

```diff
@@ -8,9 +8,25 @@
 UI_DIR = os.path.join(BASEDIR, "selfdrive", "ui")
 TRANSLATIONS_DIR = os.path.join(UI_DIR, "translations")
 LANGUAGES_FILE = os.path.join(TRANSLATIONS_DIR, "languages.json")
+TRANSLATIONS_INCLUDE_FILE = os.path.join(TRANSLATIONS_DIR, "generated_translations.h")
 
 
+def generate_translations_include():
+  # offroad alerts
+  content = "// THIS IS AN AUTOGENERATED FILE, PLEASE EDIT alerts_offroad.json\n"
+  with open(os.path.join(BASEDIR, "selfdrive/controls/lib/alerts_offroad.json")) as f:
+    for alert in json.load(f).values():
+      text = alert['text'].replace('\n', '\\n')
+      content += f'QT_TRANSLATE_NOOP("OffroadAlert", "{text}");\n'
```

Comment:
```suggestion
      content += f'QT_TRANSLATE_NOOP("OffroadAlert", R"({alert["text"]})");\n'
```

might as well use raw strings if we're not replacing quotes any more just in case

Reply:
it might  better to keep \n in .ts file,what do you think?
`<source>abc \n %1</source>`

```
<source>abc 
 %1</source>

```

Reply:
I think QT already strips them out and just puts the newlines in the ts files. They expect you to use their tool QT Linguist.

Reply:
![image](https://github.com/commaai/openpilot/assets/25857203/487db10e-32d1-4a3b-83d5-b88c376838d9)